website/docs/d/datasource_google_service_account_id_token.html.markdown

```diff
@@ -0,0 +1,105 @@
+---
+subcategory: "Cloud Platform"
+layout: "google"
+page_title: "Google: google_service_account_id_token"
+sidebar_current: "docs-google-service-account-id-token"
+description: |-
+  Produces OpenID Connect token for service accounts
+---
+
+# google\_service\_account\id\_token
+
+This data source provides a google OpenID Connect (`oidc`) `id_token`.  Tokens issued using this datasource are typically used to call external services that accept OIDC tokens for authentication (e.g. [Google Cloud Run](https://cloud.google.com/run/docs/authenticating/service-to-service)). 
+
+For more information see
+[OpenID Connect](https://openid.net/specs/openid-connect-core-1_0.html#IDToken).
+
+## Example Usage - ServiceAccount JSON credential file.
+  Datasource will the service account provided by `GOOGLE_APPLICATION_CREDENTIALS` or `GOOGLE_CLOUD_KEYFILE_JSON` environment variable. 
+  
+  ```hcl
+  data "google_service_account_id_token" oidc {
+    target_audience = "https://foo.bar/"
+  }
+
+  output "oidc_token" {
+    value = data.google_service_account_id_token.oidc.id_token
+  }
+  ```
+
+## Example Usage - `Application Default Credentials`
+  Datasource will automatically use credentials provided by [Application Default Credentials](https://cloud.google.com/docs/authentication/production)
+
+## Example Usage - Service Account Impersonation.
+  Datasource will use background impersonated credentials provided by [google_service_account_access_token](https://www.terraform.io/docs/providers/google/d/datasource_google_service_account_access_token.html).  
+
+  Note: to use the following, you must grant `target_service_account` the
+  `roles/iam.serviceAccountTokenCreator` role on itself.
+
+  ```hcl
+  data "google_service_account_access_token" "impersonated" {
+    provider = google
+    target_service_account = "impersonated-account@project.iam.gserviceaccount.com"
+    delegates = []
+    scopes = ["userinfo-email", "cloud-platform"]
+    lifetime = "300s"
+  }
+
+  provider google {
+    alias  = "impersonated"
+    access_token = data.google_service_account_access_token.impersonated.access_token
+  }
+
+  data "google_service_account_id_token" oidc {
+    provider = google.impersonated
+    target_service_account = "impersonated-account@project.iam.gserviceaccount.com"
+    delegates = []
+    include_email = true
+    target_audience = "https://foo.bar/"
+  }
+
+  output "oidc_token" {
+    value = data.google_service_account_id_token.oidc.id_token
+  }
+  ```
+
+## Example Usage - Invoking Cloud Run Endpoint
+
+  The following configuration will invoke [Cloud Run](https://cloud.google.com/run/docs/authenticating/service-to-service) endpoint where the service account for Terraform has been granted `roles/run.invoker` role previously.
+
+```hcl
+
+data "google_service_account_id_token" oidc {
+  target_audience = "https://your.cloud.run.app/"
+}
+
+data "http" cloudrun {
+  url = "https://your.cloud.run.app/"
```

Comment:
Is there an existing cloud run resource that gets you this url?

Reply:
you mean like as part of the test?   
https://www.terraform.io/docs/providers/google/r/cloud_run_service.html  does create the cloud run resource here.  (eg, you use that to deploy some small webapp with authentication enabled (whcih is default i think), then use the service account the test runs as to setup iam binding (then wait maybe 60s  or so,, then use this provider to invoke that app